[IMP] inventory: update create second warehouse doc (#9064)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/inventory_management/create_a_second_warehouse.rst

   a second warehouse automatically activates the *Storage Locations* setting.

Create new warehouse
====================

To create a new warehouse, navigate to :menuselection:`Inventory app --> Configuration -->
Warehouses`, and click :guilabel:`New`. Doing so opens a blank warehouse creation form.

Fill out the following fields as necessary:

- :guilabel:`Warehouse`: the full name of the warehouse.
- :guilabel:`Short Name`: the abbreviated code used to identify the warehouse, i.e. `WH2` (the short
  name for the default warehouse in Odoo is `WH`).
- :guilabel:`Company`: the company that owns the warehouse. This can be set as the company that owns
  the Odoo database or the company of a customer or vendor.
- :guilabel:`Address`: the address where the warehouse is located. By default, this is set to the
  company's address in the database.

.. image:: create_a_second_warehouse/create-a-second-warehouse-new-warehouse-settings.png
   :align: center
   :alt: Filled out warehouse name and address settings on warehouse creation form.

Under the :guilabel:`Warehouse Configuration` tab, edit the :guilabel:`Shipments` and

> :guilabel:`Resupply` settings as necessary.

Author: Suggestion implemented ✅

.. important::
   The :guilabel:`Shipments` and :guilabel:`Resupply` sections only appear if the
   :guilabel:`Multi-Step Routes` checkbox is enabled in :menuselection:`Inventory app -->
   Configuration --> Settings`, under the :guilabel:`Warehouse` heading. For more information about
   routes and how they work in Odoo, see :ref:`Using Routes and Pull/Push Rules <use-routes>`.

In the :guilabel:`Shipments` section, edit the following settings:

- :guilabel:`Incoming Shipments`: default incoming route to follow. Choose from one, two, and
  three-step receipts.
- :guilabel:`Outgoing Shipments`: default outgoing route to follow. Choose from one, two, and

>   three-step deliveries.

Author: Suggestion implemented ✅

In the :guilabel:`Resupply` section, edit the following settings:

> - :guilabel:`Dropship Subcontractors`: dropship subcontractors with components.

Author: Suggestion implemented ✅

- :guilabel:`Resupply Subcontractors`: allow subcontractors to be resupplied with goods from this
  warehouse.
- :guilabel:`Manufacture to Resupply`: when products are manufactured, they can be manufactured in
  this warehouse.
- :guilabel:`Manufacture`: the route that should be followed when manufacturing goods inside
  the warehouse. Choose from one, two, and three-step production.
- :guilabel:`Buy to Resupply`: when products are purchased, they can be delivered to this warehouse.
- :guilabel:`Resupply From`: tick the checkbox next to the warehouse that this warehouse should be

>   resupplied from. Resupply routes are automatically created for the selected warehouse.

Reviewer: I'm not familiar with the resupply from another warehouse route, so this section was confusing to me. Given that the resupply route is only available as a field when there's more than one warehouse created in a database, I think it's important to expand on this field a little more.

(For example, when I only have 1 WH, the "Resupply From" field isn't there:
![image](https://github.com/odoo/documentation/assets/23611711/23b824b5-37f4-4590-b78f-f45eb0e7d019)
 )
 
 Can we possibly:
 - elaborate a little more what a "warehouse that this warehouse should be resupplied from", means?
   - e.g. "Check the box next to the **other** warehouses from which **this** warehouse can replenish products. If the current warehouse does not have the stock, it can also pull products from these selected warehouses in the database. Automatic resupply routes will be created accordingly."
 - link to the [resupply WH doc](https://www.odoo.com/documentation/17.0/applications/inventory_and_mrp/inventory/warehouses_storage/inventory_management/resupply_warehouses.html) for more info

Author: Suggestion implemented ✅

.. image:: create_a_second_warehouse/create-a-second-warehouse-warehouse-configuration-tab.png
   :align: center
   :alt: Filled out Warehouse Configuration tab settings on warehouse creation form.

Once ready, click :guilabel:`Save` to save the new warehouse's settings.

Add inventory to a new warehouse
================================

> If a new warehouse is created with existing inventory in it, the inventory count should be added to
> Odoo, so the stock listed in the Odoo database reflects the real stock count in the physical
> warehouse.

Reviewer: The current explanation does a good job of explaining what we need to do in Odoo, but we can reword it to elaborate on *why* we need to adjust the stock in the first place.

```suggestion
If a new warehouse is created to represent a real-world storage location with existing stock, update the warehouse's stock using an inventory adjustment.
```

Author: Suggestion implemented ✅

Inventory can be added to a new warehouse via an inventory adjustment. Navigate to
:menuselection:`Inventory app --> Operations --> Physical Inventory`, and click :guilabel:`New`.
Doing so adds a new line at the bottom of the list of inventory adjustments.

Configure the line by filling out the following fields as necessary:

- :guilabel:`Location`: the location where the product is currently stored in the new warehouse.
  This can be set as the overall warehouse, or a location within the warehouse.

> - :guilabel:`Product`: the product being added to inventory.

Reviewer: This section on defining the fields in the Inventory adjustment page is really helpful! However, it seems like we might have some duplicate information in our documentation, since the [Inventory Adjustments doc](https://www.odoo.com/documentation/17.0/applications/inventory_and_mrp/inventory/warehouses_storage/inventory_management/count_products.html) already covers this in great detail. Maybe we could streamline it a bit to avoid redundancy? 

We can either replace this and the following lines with a "seealso" admonition block linking to the Inventory Adjustments page, and/or summarize what we can do on this page.

Author: I understand the logic here, but I feel like the process being explained here isn't a bad thing, especially considering it could be part of the workflow for creating a second WH. I did ad a seealso block for that doc, but I'm gonna keep this section as is for now. If I receive the same feedback again, I'll consider changing it

Create a second warehouse
=========================

In Odoo *Inventory*, a *warehouse* is a physical building or space where items are stored. By

> default, each Odoo database has one warehouse already pre-configured, with the address set as
> the company's address.

Reviewer 2: Early break :)
```suggestion
default, each Odoo database has one warehouse already pre-configured, with the address set as the
company's address.
```

Author: Suggestion implemented ✅

Under the :guilabel:`Warehouse Configuration` tab, configure the warehouse's main workflow in the
:guilabel:`Shipments` section, and set available replenishment methods in the :guilabel:`Resupply`
section.

.. important::
   The :guilabel:`Shipments` and :guilabel:`Resupply` sections only appear if the
   :guilabel:`Multi-Step Routes` checkbox is enabled in :menuselection:`Inventory app -->
   Configuration --> Settings`, under the :guilabel:`Warehouse` heading. For more information about
   routes and how they work in Odoo, see :ref:`Using Routes and Pull/Push Rules <use-routes>`.

In the :guilabel:`Shipments` section, choose between one, two, or three-step processes for both
incoming and outgoing shipments.

In the :guilabel:`Resupply` section, edit the following settings:

- :guilabel:`Dropship Subcontractors`: enable this feature to coordinate with vendors, and have
  components dropshipped directly to subcontractors for manufacturing.
- :guilabel:`Resupply Subcontractors`: allow subcontractors to be resupplied with goods from this
  warehouse.
- :guilabel:`Manufacture to Resupply`: when products are manufactured, they can be delivered to this
  warehouse.

> - :guilabel:`Manufacture`: the route that should be followed when manufacturing goods inside
>   the warehouse. Choose from one, two, and three-step production.

Reviewer 2: ![another one](https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExNHpocmtwYjQ0Nms2Ym1oMnFteTZwejVjNGduanpreXo4cmZzeW5icCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l0HlQ7LRalQqdWfao/giphy.gif)
```suggestion
- :guilabel:`Manufacture`: the route that should be followed when manufacturing goods inside the
  warehouse. Choose from one, two, and three-step production.
```

Author: Suggestion implemented ✅

- :guilabel:`Address`: the address where the warehouse is located. By default, this is set to the
  company's address in the database.

.. image:: create_a_second_warehouse/create-a-second-warehouse-new-warehouse-settings.png
   :align: center
   :alt: Filled out warehouse name and address settings on warehouse creation form.

Under the :guilabel:`Warehouse Configuration` tab, configure the warehouse's main workflow in the
:guilabel:`Shipments` section, and set available replenishment methods in the :guilabel:`Resupply`
section.

.. important::
   The :guilabel:`Shipments` and :guilabel:`Resupply` sections only appear if the
   :guilabel:`Multi-Step Routes` checkbox is enabled in :menuselection:`Inventory app -->
   Configuration --> Settings`, under the :guilabel:`Warehouse` heading. For more information about
   routes and how they work in Odoo, see :ref:`Using Routes and Pull/Push Rules <use-routes>`.

In the :guilabel:`Shipments` section, edit the following settings:

- :guilabel:`Incoming Shipments`: choose between one, two, or three-step processes for incoming
  packages.
- :guilabel:`Outgoing Shipments`: choose between one, two, or three-step processes for outgoing

>   packages.

Author: Suggestion implemented ✅

In the :guilabel:`Resupply` section, edit the following settings:

- :guilabel:`Dropship Subcontractors`: enable this feature to coordinate with vendors, and have
  components dropshipped directly to subcontractors for manufacturing.
- :guilabel:`Resupply Subcontractors`: allow subcontractors to be resupplied with goods from this
  warehouse.
- :guilabel:`Manufacture to Resupply`: when products are manufactured, they can be manufactured in

>   this warehouse.

Author: You are right, this is a typo - I must have looked it over when writing it. Nice catch, adjusting accordingly - Suggestion implemented ✅

- :guilabel:`Manufacture`: the route that should be followed when manufacturing goods inside
  the warehouse. Choose from one, two, and three-step production.
- :guilabel:`Buy to Resupply`: when products are purchased, they can be delivered to this warehouse.
- :guilabel:`Resupply From`: Check the box next to the other warehouses from which this warehouse
  can replenish products. If the current warehouse does not have the stock, it can also pull
  products from these selected warehouses in the database. Automatic resupply routes will be created
  accordingly.

.. image:: create_a_second_warehouse/create-a-second-warehouse-warehouse-configuration-tab.png
   :align: center
   :alt: Filled out Warehouse Configuration tab settings on warehouse creation form.

Once ready, click :guilabel:`Save` to save the new warehouse's settings.

Add inventory to a new warehouse
================================

If a new warehouse is created to represent a real-world storage location with existing stock, update
the warehouse's stock using an inventory adjustment.

.. seealso::
   :doc:`count_products`

> Inventory can be added to a new warehouse via an inventory adjustment. Navigate to

Reviewer 3: This first sentence seems redundant to the section's introductory sentence and doesn't transition well afterwards, I don't think. Here's a suggestion to help ease that transition into the direct action we're instructing on the UI.

```suggestion
To perform an inventory adjustment, navigate to
```

Author: Nice catch, it is redundant. Suggestion implemented ✅

:menuselection:`Inventory app --> Operations --> Physical Inventory`, and click :guilabel:`New`.
Doing so adds a new line at the bottom of the list of inventory adjustments.

Configure the line by filling out the following fields as necessary:

- :guilabel:`Location`: the location where the product is currently stored in the new warehouse.
  This can be set as the overall warehouse, or a location within the warehouse.
- :guilabel:`Product`: the product being added to inventory.
- :guilabel:`Lot/Serial Number`: the lot that the product belongs to or the serial number used to
  identify it, if the product is tracked.
- :guilabel:`On Hand Quantity`: the total quantity of the product stored in the location for which

>   inventory is being adjusted. For a new warehouse or location, this should be set to `0.00`.

Reviewer 3: Consider adding more context around the conditions for the value we're suggesting.
 
It's possible that someone IRL did a physical count at the new warehouse (or new ahead of time), and inputted it into the new warehouse in Odoo before we (i.e., the office admin) performed an "official" batch inventory adjustment.

```suggestion
  inventory is being adjusted. For a new warehouse or location where a count or a prior inventory
  adjustment has not taken place, this should be set to `0.00`.
```

- :guilabel:`UoM`: the unit of measure (UoM) used for counting the product.
- :guilabel:`Counted Quantity`: the amount of the product being added to inventory.
- :guilabel:`Difference`: the difference between the on-hand and counted quantities. This
  automatically updates to reflect the value entered in the :guilabel:`Counted Quantity` column.
- :guilabel:`Scheduled Date`: the date selected for the next inventory count for this product.
- :guilabel:`User`: the user who recorded the inventory adjustment in the database.

> .. image:: create_a_second_warehouse/create-a-second-warehouse-inventory-adjustment.png

Reviewer 3: This screenshot is a bit small and has both x and y axis scrolls visible. Consider retaking up to 933px width so we really capture those details for the reader, as well as at a good crop. It would also be nice to see the column header labels so we know what we're looking at.

   :align: center
   :alt: Filled out inventory adjustment line for product in new warehouse.

Once each line for the products being added to the new warehouse is configured, click
:guilabel:`Apply` on each line to apply the new inventory counts.

The values in the :guilabel:`On Hand Quantity` column update to reflect those in the
:guilabel:`Counted Quantity` column, and the products added appear in the new warehouse's stock.

.. note::
   If products tracked using a lot number or serial number are added without specifying the tracking
   number, a :guilabel:`Tracked Products in Inventory Adjustment` pop-up window appears upon
   clicking :guilabel:`Apply` on that product's line.

   The pop-up indicates tracked products were added, but the lots/serial numbers were not defined.
   To apply the adjustment without the numbers, click :guilabel:`Apply`. To add the numbers before
   applying, click :guilabel:`Discard`, and add the numbers. Then, once ready, click
   :guilabel:`Apply` again.

>    .. image:: create_a_second_warehouse/create-a-second-warehouse-popup-window.png

Reviewer 3: Consider capturing the full pop-up window here, as only the right and bottom borders are visible.

Author: Retook screenshot and replaced with better image with all borders visible
